ee/fleetd-chrome/src/tables/Table.ts

```diff
@@ -0,0 +1,149 @@
+// This is the foundation for all of the tables. We take the low-level SQLite functions and map them
+// to an abstraction so that table implementations only need to define their name, columns, and
+// generate() function.
+
+import * as SQLite from "wa-sqlite";
+
+class cursorState {
+  rowIndex: number;
+  rows: Record<string, string | number>[];
+  error: any;
+}
+
+export default abstract class Table implements SQLiteModule {
+  sqlite3: SQLiteAPI;
+  db: number;
+  name: string;
+  columns: string[];
+  cursorStates: Map<number, cursorState>;
```

Comment:
nice use of `Map` for `this.cursorStates`. It's preferred because of its performance when dealing with mutations of a key/value store. 👍🏽